### PR TITLE
generate make-dist archives with github actions

### PR DESCRIPTION
**Pull request by aws-nslick**
*Issue #, if available:* #719

*Description of changes:* Take a human out of the loop when generating pre-autotool'd tarballs. Can extend this action with a second stage that unpacks the resulting tarball and enforces that the expected PACKAGE_VERSION for the tag is represented in all paths. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by Zhenye-Na** [comment]
bot:aws:retest

**Review by rauteric — APPROVED** [review]

File: .github/workflows/tag-makedist.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        efainstallerdir:
+          - ALINUX2023
+        nvidiadistro:
+          - amzn2023
+        configmanager:
+          - dnf config-manager
+        cudapackages:
+          - cuda-cudart-devel-12-6 cuda-crt-12-6
+
+    runs-on: ubuntu-latest
```
[the remark this answers is not in the document]
Comment (rajachan) [reply .github/workflows/tag-makedist.yaml:28]: Because the container is al2023.

File: .github/workflows/tag-makedist.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  workflow_dispatch:
+  push:
+    tags:
+      - release/**
+      - v**
+
+jobs:
+  amazonlinux:
+    strategy:
+      matrix:
```
Comment (rajachan) [inline .github/workflows/tag-makedist.yaml:12]: Why do these need to be a matrix?

**Review by rajachan — APPROVED** [review]

File: .github/workflows/tag-makedist.yaml (hunk 2 above)
Comment (aws-nslick) [reply .github/workflows/tag-makedist.yaml:12]: I don't like using environment variables in github actions because of quoting issues.

**Pull request merged**